app/src/ui/add-repository/add-existing-repository.tsx

```diff
@@ -158,11 +183,18 @@ export class AddExistingRepository extends React.Component<
 
     const path = directory[0]
     const isRepository = await isGitRepository(path)
+    const isBareRepositoryResult = await checkIfRepositoryIsBare(path)
```

Comment:
:art: I think we can distill this down into a one-liner, based on what I think `null` represents:

```ts
const isBareRepository = (await checkIfRepositoryIsBare(path)) || false
```